komodo/symlink/suggester/release.py

```diff
@@ -1,47 +1,47 @@
+import datetime
 import ntpath
 import os
 import re
-from datetime import datetime
 
 
 def path_leaf(path):
     head, tail = ntpath.split(path)
     return tail or ntpath.basename(head)
 
 
-def diff_month(d1, d2):
-    return (d1.year - d2.year) * 12 + d1.month - d2.month
+def diff_month(date_1: datetime.date, date_2: datetime.date) -> int:
+    return (date_1.year - date_2.year) * 12 + date_1.month - date_2.month
 
 
-class Release(object):
-    def __init__(self, release_id):
+class Release:
+    def __init__(self, release_id: str):
         self.release_id = release_id
 
-    def __repr__(self):
+    def __repr__(self) -> str:
```

Comment:
It's not always possible, but I think in general the `__repr__` should be something you could `eval`. So for example: `return f"Release({self.release_id})"`

The methods that want to use `repr(self)` etc seem like they could just use `len(self.release_id)`, which seems more explicit to me. Indeed, in `is_concrete()` I feel like a regex would be even more explicit.

IDK, not a big deal, just made me double-take.

Reply:
🤔 Should the type of the returned object be a Callable?

Reply:
Maybe just leave it untyped

Reply:
Should be `str`